### PR TITLE
make docker-compose file work?

### PR DESCRIPTION
**Pull request by scyto**
the build context breaks this file for those who just want to pull this yaml file and then do docker-compose up - suggesting we remove it unless there is some reason not to?
the network mode of host breaks on my latest docker hosts as ports definition is incompatible with host mode (apparently) - suggest we change to bridge unless there is some reason not to?

(aka i am pretty certain buildx is using the dockerfile directly and not this docker-compose)

### CODE REVIEW

**Comment by scyto** [comment]
closing, going to refork everything and test

**Comment by dulitz** [comment]
Network mode host is needed in order for infinitude to open a connection to a networked RS-485 bridge, and it is needed in order to connect to external services such as Carrier's cloud service.

**Comment by scyto** [comment]
> Network mode host is needed in order for infinitude to open a connection to a networked RS-485 bridge, and it is needed in order to connect to external services such as Carrier's cloud service.

no it isn't i have done both today just fine using bridge (modulo the 404 issue)

**Comment by nebulous** [comment]
Do we need to spell these out explicitly at all I wonder? The. best line of code/config is no code/config after all.

**Comment by dulitz** [comment]
Let's return to this question after scyto figures out why the container isn't working.

If you use the default bridge, it is _possible_ to [configure it to be non-isolated](https://docs.docker.com/network/bridge/#enable-forwarding-from-docker-containers-to-the-outside-world), and apparently scyto has done this, but it is not the default.

If you use a user-defined bridge and use the right set of options, you might be able to gain access to the outside world without the user needing to do special config outside of Docker. I don't know how, but there's probably a way.

Host mode avoids those issues -- the best config is no config -- but it's all moot since the container doesn't work. :)

**Comment by dulitz** [comment]
BTW I looked at a lot of actual docker installations (as opposed to the docker docs), and it seems that all of them behave like @scyto describes: containers have access to the external network.

**Comment by scyto** [comment]
There is a nuance in docker that most folks miss.

The default bridge and a user created bridge do *not* behave in the same way.

For example the default bridge can have issues with DNS across containers on the same network (cant ping other containers, makes people think DNS is borked and you see people do all sorts of recommendations about changing the config files in the container).

The correct way (because the –link option is deprecated) is to always create a custom bridge for containers to live on – then DNS works correctly both inside the custom bridge container for container to container addressing by name and also sometimes solves DNS external issues too.

Use bridge networks | Docker Documentation<https://docs.docker.com/network/bridge/#differences-between-user-defined-bridges-and-the-default-bridge>

Also both bridge types are NAT’d this means that the containers won’t see any form of broadcast traffic (e.g. mDNS from outside the container network) and if a port is not published then nothing can reach from your home network into the container by default.

Tl;dr difference in bridge functionality can make one think there is no external connectivity 😊

Took me hundreds of hours of messing about to get to the bottom of bridge vs default bridge vs host vs macvlan, they didn’t make it obvious or easy ☹
